src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,46 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+    private static final String COMMA = ",";
+    private static final String COLON = ":";
+    private static final String VERTICAL_LINE = "|";
+
+    public static int splitAndSum(String text) {
+        if (checkEmptyOrNullString(text)) {
+            return 0;
+        }
+
+        String[] numbers = splitNumber(text);
+
+        checkAllDigit(numbers);
+
+        return sum(numbers);
+    }
+
+    private static String[] splitNumber(String text) {
+        Matcher matcher = Pattern.compile("//(.)\n(.*)").matcher(text);
+        if (matcher.find()) {
+            String delimiter = matcher.group(1);
+            return matcher.group(2).split(delimiter);
+        }
+        return text.split(COMMA + VERTICAL_LINE + COLON);
+    }
+
+    private static boolean checkEmptyOrNullString(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static void checkAllDigit(String[] numbers) {
+        if (!Arrays.stream(numbers).allMatch(n -> n.matches("\\d+"))) {
+            throw new RuntimeException();
+        }
+    }
+
+    private static int sum(String[] numbers) {
+        return Arrays.stream(numbers).mapToInt(Integer::parseInt).sum();
```

Comment:
```suggestion
        return Arrays.stream(numbers)
        .mapToInt(Integer::parseInt)
        .sum();
```

스트림은 아래로 내려 쓰는 것이 디버깅 및 가독성에 좋은 것 같아요.